index.html

```diff
@@ -389,9 +393,48 @@ <h2>
                     <var>message</var>)</code>.</p>
                   </li>
                   <li>
-                    <p><a>Resolve</a> <var>p</var> with <var>stream</var> and
-                    abort these steps. This invocation of {{MediaDevices/getDisplayMedia()}}
-                    is now considered to have produced a new <dfn>capture-session</dfn></p>
+                    <p>
+                      This invocation of {{MediaDevices/getDisplayMedia()}} is now
+                      considered to have produced a new <dfn>capture-session</dfn>.
+                    </p>
+                  </li>
+                  <li>
+                    If <var>controller</var> is not <code>null</code>:
+                    <ol>
+                      <li>
+                        <p>
+                          Set <var>controller</var>.{{CaptureController/[[Source]]}}
+                          to <var>stream</var>'s video track's <a data-cite="GETUSERMEDIA#dfn-source-0">[[\Source]]</a>.
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          Set <var>controller</var>.{{CaptureController/[[DisplaySurfaceType]]}} to the 
+                          to <var>stream</var>'s video track's {{DisplayCaptureSurfaceType}}.
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          Schedule a <dfn>task to close the window of opportunity to set focus behavior</dfn>,
+                          which will set <var>controller</var>.{{CaptureController/[[FocusWindowClosed]]}}
+                          to <code>true</code> when it executes.
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          If the <a>current settings object</a>'s [=relevant global object=]'s
+                          [=associated `Document`=] ever <a data-cite="!HTML/#focused">loses focus</a>,
```

Comment:
If you could suggest a good place to move it, I'm happy to oblige. But note that we need to somehow clarify that this condition only comes into effect *after* the user makes their selection. That is, we don't count as `InitialFocusLost` cases were the user interacts with another tab, native app, or OS-level widget, if that happens **before** they finish their interaction with the picker. This is why this step was put here originally. Do you have an idea for how to express that better?

Reply:
I see, yeah I wouldn't infer that solely from its placement here. I would place it right after these algorithm steps as a new paragraph with something like:

"When the top-level document loses focus, run the following steps on all CaptureController objects in that document and in documents of its nested browsing contexts:

1. If [[Source]] is null, abort these steps.
2. Set [[InitialFocusLost]] to true.

Would that work?

Reply:
But how does this work with UAs that defocus ahead of the prompt (like Chrome does?)

Reply:
Suggestion adopted.

Reply:
I wonder if we're in user agent territory here since some browsers are focused in this case and others aren't. What's the behavior we're protecting here? Is the idea that we don't want to compete with a user-driven focus change that might have happened in the interim?